### PR TITLE
Completed HW1

### PR DESCRIPTION
**Pull request by CaseyH19**
Casey Hittinger
43199706

### CODE REVIEW

File: ex01.ex [hunk 1]
```diff
@@ -86,7 +86,9 @@ defmodule Ex01 do
   # function. The examples below will make this clearer :)               #
   ########################################################################
 
-  create_adder = your_anonymous_function(1)
+  create_adder = fn (val) ->
```
Comment (pragdave) [inline ex01.ex:89]: Nice layout!

File: ex02.ex [hunk 2]
```diff
@@ -41,7 +41,7 @@ defmodule Ex02 do
   # Do the same using the & syntax #
   ##################################
 
-  list2b = your_anonymous_function(1, 2)
+  list2b = &([&1 + &2, &1 - &2])
```
Comment (pragdave) [inline ex02.ex:44]: Just an aside: you don't need the parens here: `&[ &1 + &2, ...]`

File: ex03.ex [hunk 3]
```diff
@@ -59,7 +59,16 @@ defmodule Ex03 do
 
   """
 
-  def odd_even . . . "your code"
+  def odd_even([]), do: []
+  def odd_even([ h | t ]) do
+    [ res(Integer.is_even(h))| odd_even(t)]
+  end
+  def res(false) do
```
Comment (pragdave) [inline ex03.ex:66]: No deduction, but `res`??? Really???

File: ex03.ex [hunk 4]
```diff
@@ -80,8 +89,13 @@ defmodule Ex03 do
       true
 
   """
+  def list_contains([], _val), do: false
```
Comment (pragdave) [inline ex03.ex:92]: # -1

This is just a little too convoluted. How about

~~~ elixir
  def list_contains([], _value),             do: false
  def list_contains([value | _rest], value), do: true
  def list_contains([_ | rest], value),      do: list_contains(rest, value)
~~~

File: ex03.ex [hunk 5]
```diff
@@ -104,8 +118,14 @@ defmodule Ex03 do
       false
 
   """
-
-  def list_equal . . . "your code"
+  def list_equal([], []), do: true
```
Comment (pragdave) [inline ex03.ex:121]: # -1

Same comment here. It could just be:

~~~ elixir
  def list_equal([], []),               do: true
  def list_equal([ h | t1 ], [h | t2]), do: list_equal(t1, t2)
  def list_equal(_, _),                 do: false
~~~

File: ex03.ex [hunk 6]
```diff
@@ -153,7 +173,15 @@ defmodule Ex03 do
   Think a little about a nice way to lay this code out.
   """
 
-  def won . . . "your code"
```
Comment (pragdave) [inline ex03.ex:156]: # -1

I was hoping for a 2d layout

~~~ elixir
 def won({ m, m, m,
            _, _, _,
            _, _, _ }), do: m

  def won({ _, _, _,
            m, m, m,
            _, _, _ }), do: m
# . . .
~~~

File: ex04.ex [hunk 7]
```diff
@@ -39,7 +39,9 @@ defmodule Ex04 do
       [ 1, 2, 3, 4, 5 ]
 
   """
-  def reverse . . . "your code"
+  def reverse([]), do: []
```
Comment (pragdave) [inline ex04.ex:42]: # -1

There's no need to test for an empty list, because reduce already handles it.

File: ex04.ex [hunk 8]
```diff
@@ -75,8 +78,10 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
-
+  def even_odd([]), do: nil
```
Comment (pragdave) [inline ex04.ex:81]: # +1

So that's clever: reversing the input list means you don't have to reverse both result lists.

That saved you from getting a deduction for line 82, which has way too much going on...

**Comment by pragdave** [comment]
# +2

for doing lots of commits during development
